includes/DataObjects/User.php

```diff
@@ -644,7 +644,7 @@ public function isAdmin()
     
 	public function isCheckuser()
 	{
-		return $this->checkuser == 1 || $this->oauthCanCheckUser();
+		return $this->checkuser == 1 || ($this->isOAuthLinked() && $this->oauthCanCheckUser());
```

Comment:
It's not a recursion fix at all, it's a fix for when the user is *not* linked to OAuth (and when that's not forced, such as development systems), it still does the OAuth call to get the groups and see if checkuser is one of them. I've been unable to reproduce it on my system (I use OAuth), but I spotted it while debugging. `isOAuthLinked()` returns a false if OAuth isn't configured for that user, thus this is simply short-circuiting the OAuth check.